mirrord/layer/src/detour.rs

```diff
@@ -6,16 +13,52 @@ use std::{cell::RefCell, ops::Deref, os::unix::prelude::*, path::PathBuf};
 
 use crate::error::HookError;
 
-thread_local!(pub(crate) static DETOUR_BYPASS: RefCell<bool> = RefCell::new(false));
+thread_local!(
+    /// Holds the thread-local state for bypassing the layer's detour functions.
+    ///
+    /// ## Warning
+    ///
+    /// Do **NOT** use this directly, instead use `DetourGuard::new` if you need to
+    /// create a bypass inside a function (like we have in
+    /// [`TcpHandler::create_local_stream`](crate::tcp::TcpHandler::create_local_stream)).
+    ///
+    /// Or rely on the [`hook_guard_fn`](mirrord_layer_macro::hook_guard_fn) macro.
+    ///
+    /// ## Details
+    ///
+    /// Some of the layer functions will interact with [`libc`] functions that we are hooking into,
+    /// thus we could end up _stealing_ an actual call that we want to make. An example of this
```

Comment:
```suggestion
    /// thus we could end up _stealing_ a call by the layer itself rather than by the binary the layer is injected into. An example of this
```